src/platforms/esp32/main/socket_driver.c

```diff
@@ -192,91 +177,236 @@ static term do_bind(Context *ctx, term params)
+    int event_descriptor = open_event_descriptor(conn);
+
+    term ret = do_bind(ctx, params);
+
+    socket_data->active = active;
+    if (ret == OK_ATOM && active == TRUE_ATOM) {
+        EventListener *listener = malloc(sizeof(EventListener));
+        if (IS_NULL_PTR(listener)) {
+            fprintf(stderr, "Failed to allocate memory: %s:%i.\n", __FILE__, __LINE__);
+            abort();
+        }
+        listener->fd = event_descriptor;
+        listener->expires = 0;
+        listener->expiral_timestamp.tv_sec = INT_MAX;
+        listener->expiral_timestamp.tv_nsec = INT_MAX;
+        listener->one_shot = 0;
+        listener->data = ctx;
+        listener->handler = active_recvfrom_callback;
+        linkedlist_append(&ctx->global->listeners, &listener->listeners_list_head);
+        socket_data->active_listener = listener;
+        TRACE("socket: initialized\n");
+    } else {
         return port_create_error_tuple(ctx, BADARG_ATOM);
     }
-    term proto = interop_proplist_get_value(params, PROTO_ATOM);
+    return ret;
+}
 
-    if (term_is_nil(proto)) {
+static term do_connect(SocketDriverData *socket_data, Context *ctx, term address, term port)
+{
+    // TODO handle IP addresses
+    if (!term_is_list(address) || !term_is_list(port)) {
         return port_create_error_tuple(ctx, BADARG_ATOM);
     }
 
-    enum netconn_type conn_type;
+    int ok;
+    char *addr_str = interop_term_to_string(address, &ok);
+    if (!ok) {
+        return port_create_error_tuple(ctx, BADARG_ATOM);
+    }
+    TRACE("socket_driver: resolving %s ...\n", addr_str);
 
-    if (proto == UDP_ATOM) {
-        conn_type = NETCONN_UDP;
+    struct ip_addr remote_ip; 
+    err_t status = netconn_gethostbyname(addr_str, &remote_ip);
 
-        term controlling_process = interop_proplist_get_value_default(params, CONTROLLING_PROCESS_ATOM, term_invalid_term());
-        if (!(term_is_invalid_term(controlling_process) || term_is_pid(controlling_process))) {
-            TRACE("invalid controlling process pid");
-            return port_create_error_tuple(ctx, BADARG_ATOM);
-        }
-        socket_data->controlling_process = controlling_process;
+    free(addr_str);
 
-        term binary = interop_proplist_get_value_default(params, BINARY_ATOM, TRUE_ATOM);
-        if (!(binary == TRUE_ATOM || binary == FALSE_ATOM)) {
-            TRACE("invalid binary flag");
-            return port_create_error_tuple(ctx, BADARG_ATOM);
-        }
-        socket_data->binary = binary;
+    if (status != ERR_OK) {
+        // TODO atom
+        return port_create_sys_error_tuple(ctx, UNDEFINED_ATOM, status);
+    }
 
-        term buffer = interop_proplist_get_value_default(params, BUFFER_ATOM, term_from_int(BUFSIZE));
-        if (!term_is_integer(buffer)) {
-            TRACE("invalid buffer flag");
-            return port_create_error_tuple(ctx, BADARG_ATOM);
-        }
-        socket_data->buffer = buffer;
+    char *port_str = interop_term_to_string(port, &ok);
```

Comment:
are you ignoring ok here too? It would be better either to check it or leave a TODO here

Reply:
See above.  All ints, all the time.